### ISSUE TITLE
Unable to build function using Go runtime

### ISSUE THREAD

**Issue by MtthwBrwng** [aws-amplify/amplify-cli#11306]
### Before opening, please confirm:

- [X] I have installed the latest version of the Amplify CLI (see above), and confirmed that the issue still persists.
- [X] I have [searched for duplicate or closed issues](https://github.com/aws-amplify/amplify-cli/issues?q=is%3Aissue+).
- [X] I have read the guide for [submitting bug reports](https://github.com/aws-amplify/amplify-cli/blob/dev/CONTRIBUTING.md#bug-reports).
- [X] I have done my best to include a minimal, self-contained set of instructions for consistently reproducing the issue.
- [X] I have removed any sensitive information from my code snippets and submission.

### How did you install the Amplify CLI?

npm

### If applicable, what version of Node.js are you using?

v18.10.0

### Amplify CLI Version

10.4.0

### What operating system are you using?

Windows (10.0.19045 Build 19045)

### Did you make any manual changes to the cloud resources managed by Amplify? Please describe the changes made.

No manual changes made

### Amplify Categories

function

### Amplify Commands

build

### Describe the bug

On a fresh project building a function using the Go runtime on CLI version > 10.2.3 results in the error below regardless of Go version

```
Error: Error installing build-lambda-zip: TypeError: Invalid Version: 1.17
    at winZip (C:\snapshot\repo\build\node_modules\amplify-go-function-runtime-provider\lib\runtime.js:154:15)
    at packageResource (C:\snapshot\repo\build\node_modules\amplify-go-function-runtime-provider\lib\runtime.js:137:15)
    at processTicksAndRejections (internal/process/task_queues.js:95:5)
    at async packageFunction (C:\snapshot\repo\build\node_modules\amplify-category-function\lib\provider-utils\awscloudformation\utils\packageFunction.js:47:27)
    at async Object.run (C:\snapshot\repo\build\node_modules\amplify-category-function\lib\commands\function\build.js:23:13)
    at async Object.executeAmplifyCommand (C:\snapshot\repo\build\node_modules\amplify-category-function\lib\index.js:267:5)
    at async executePluginModuleCommand (C:\snapshot\repo\build\node_modules\@aws-amplify\cli-internal\lib\execution-manager.js:142:5)
    at async executeCommand (C:\snapshot\repo\build\node_modules\@aws-amplify\cli-internal\lib\execution-manager.js:40:9)
    at async Object.run (C:\snapshot\repo\build\node_modules\@aws-amplify\cli-internal\lib\index.js:153:5)
There was an error building the function resources
```

### Expected behavior

I'm currently attempting to build using Go version 1.19, if there was an error I'd expect it to be the value provided, not 1.17. I also expect it to build for 1.17+ given the additions added in commit ac31ec020d706eabdeab0405caeea2a7e00a8ae9

### Reproduction steps

1. `amplify init`
```
? Enter a name for the project go117amplifybug
The following configuration will be applied:

Project information
| Name: go117amplifybug
| Environment: dev
| Default editor: Visual Studio Code
| App type: javascript
| Javascript framework: none
| Source Directory Path: src
| Distribution Directory Path: dist
| Build Command: npm.cmd run-script build
| Start Command: npm.cmd run-script start

? Initialize the project with the above configuration? Yes
Using default provider  awscloudformation
? Select the authentication method you want to use: AWS profile
```
2. `amplify add function`
```
PS C:\Users\Matt\Code\Sandbox\go1.17_amplify_bug> amplify add function
? Select which capability you want to add: Lambda function (serverless function)
? Provide an AWS Lambda function name: go117amplifybug3c7d1471
? Choose the runtime that you want to use: Go
Only one template found - using Hello World by default.

Available advanced settings:
- Resource access permissions
- Scheduled recurring invocation
- Lambda layers configuration
- Environment variables configuration
- Secret values configuration

? Do you want to configure advanced settings? No
? Do you want to edit the local lambda function now? No
Successfully added resource go117amplifybug3c7d1471 locally.
```
3. `amplify build function`

### Project Identifier

b02e83a93b27f19213a9fcd3cf4be3dc

### Additional information

It appears the bit of code causing the error is [amplify-go-function-runtime-provider/src/runtime.ts](https://github.com/aws-amplify/amplify-cli/blob/ac06cb27b1d9e83a09b2f7ac99571f0d8825a64e/packages/amplify-go-function-runtime-provider/src/runtime.ts#L121), it logically make sense from what I can see, maybe there is a difference between windows/linux in how the semver version is parsed resulting in one of the conditions being thrown.

`go version go1.19.2 windows/amd64`

### Workaround
Use Amplify CLI version `10.2.3` and Go version `1.16`

**Comment by josefaidt** [comment]
Hey @MtthwBrwng :wave: it appears this is due to the `gte` use at this line https://github.com/aws-amplify/amplify-cli/blob/ac06cb27b1d9e83a09b2f7ac99571f0d8825a64e/packages/amplify-go-function-runtime-provider/src/runtime.ts#L175

I copied the relevant bits to a spare script and was able to reproduce immediately

```ts
import { SemVer, coerce, gte, lt } from 'semver'

const CURRENT_VERSION_OUTPUT = 'go version go1.19.2 windows/amd64'
const BASE_VERSION = '1.17'

export const getGoVersion = (): SemVer => {
  // Validate go version
  // const versionOutput = spawnSync('go', ['version'])
  const versionOutput = CURRENT_VERSION_OUTPUT
  if (versionOutput) {
    const parts = versionOutput.split(' ')
    // Output: go version go1.14 darwin/amd64
    if (
      parts.length !== 4 ||
      !parts[2].startsWith('go') ||
      coerce(parts[2].slice(2)) === null
    ) {
      throw new Error(`Invalid version string: ${versionOutput}`)
    }
    const goVersion = <SemVer>coerce(parts[2].slice(2))
    return goVersion
  }
  throw new Error(`Invalid version string: ${versionOutput}`)
}

console.log(getGoVersion())
// { options: { }, loose: false, includePrerelease: false, raw: "1.19.2", major: 1, minor: 19, patch: 2, prerelease: [], build: [], version: "1.19.2" }

// console.log(gte(getGoVersion(), BASE_VERSION))
// errors with: TypeError: Invalid Version: 1.17

console.log(gte(getGoVersion(), coerce(BASE_VERSION)))
// true
```

From here it seems the string `'1.17'` should be modified to `coerce('1.17')`. Marking as a bug

**Comment by MtthwBrwng** [comment]
@josefaidt I've been able to dig into the CLI and make those changes and can confirm that's the issue. It only required two additions, the one you suggested, as well as adding a module version to the `go install` command. If needed I could sit down and figure out submitting a PR. If not the changes I made exist here e10fe5b0ac406b55d454ed92531e8710a963400f.

**Comment by rm-rf42** [comment]
Having the same issue.

ERROR: 
`🛑 Error installing build-lambda-zip: TypeError: Invalid Version: 1.17`

Amplify Version: 10.5.0
Go Mod: go 1.18 ( had to use due to the backend not being able to be built with 1.19 in Amplify backend server)

**Comment by MtthwBrwng** [comment]
> @MtthwBrwng + @josefaidt Any work around besides downgrading the cli to version 10.2.3? Anything I can modify locally to allow it to push?

PR has just been reviewed, hopefully here soon it'll be merged in and make a release fixing the issue.

**Comment by MtthwBrwng** [comment]
Any update on when this might be bundled into a release? Haven't been able to upgrade the CLI for the last 2 months since Go support is broken regardless of Go version on any CLI version > 10.2.3

**Comment by edwardfoyle** [comment]
This is resolved in version 10.8.0 of the CLI. Please upgrade to try it out.
